### DISCUSSION TITLE
text  label  in q-chip

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3084] [Q&A]
The method set_text() should ultimately also write to the label, right? Doesn't it have the same effect as the other one?

```py
from nicegui import ui 

cp=ui.chip('chip-text')
btn=ui.button('change label').on_click(lambda:cp.props('label="change text"'))
btn2=ui.button('change text').on_click(lambda:cp.set_text('change text'))

ui.run()
```

**Comment by falkoschindler** [comment]
Hi @v479038280,

What's wrong with `set_text`?
The `ui.chip` element uses the default slot for showing text. According to the [Quasar documentation](https://quasar.dev/vue-components/chip/), the "label" prop is an alternative way to show a label:

> Chip's content as string; overrides default slot if specified

I'd recommend not touching the "label" prop but to use `text` and/or `set_text` instead.

**Reply by v479038280** [reply]
>Chip's content as string; overrides default slot if specified

After specifying "label=change text", the default content of the chip, which is 'chip-text', is not being overridden. Instead, the two characters are being concatenated together.

By specifying props, we can modify multiple attributes at once, such as color, size, etc. In contrast, set_text is only for text. Props are preferred when multiple changes are needed.
